### PR TITLE
Video upload support

### PR DESCRIPTION
**Pull request by ross-pfahler**

### CODE REVIEW

**Comment by ross-pfahler** [comment]
needs https://github.com/Livefyre/streamhub-sdk/pull/320 and https://github.com/Livefyre/streamhub-editor/pull/5 (and then need to re-pin versions)

File: src/javascript/content-editor/view.js [hunk 1]
```diff
@@ -65,13 +76,14 @@ function handleLogout() {
  * @private
  */
 ContentEditor.prototype._addUploadButton = function () {
+    var uploadEl = this.getElementsByClass(this.classes.EDITOR_UPLOAD);
+    if (!uploadEl.length) {
+        uploadEl = $(document.createElement('div')).addClass(this.classes.EDITOR_UPLOAD);
```
Comment (markdoten) [inline src/javascript/content-editor/view.js]: Wouldn't `$('div')` do the same thing?

File: src/javascript/upload/view.js [hunk 2]
```diff
@@ -143,22 +143,65 @@ Upload.prototype._processResponse = function (err, inkBlob) {
...
-                var content = self._packageInput(convertedBlob);
-                contents.push(content);
-                self.writeToDestination(content);
+                var url = self._cacheUrl + convertedBlob.key;
+                var attachment = {
+                    type: 'photo',
+                    url: url,
+                    link: url,
+                    provider_name: 'Livefyre'
+                    //TODO (joao) images dimensions?!!!!
```
Comment (markdoten) [inline src/javascript/upload/view.js]: This probably doesn't need dimensions, right? If not, can remove the comment.

File: src/javascript/upload/view.js [hunk 3]
```diff
@@ -143,22 +143,65 @@ Upload.prototype._processResponse = function (err, inkBlob) {
...
+
+                var content = new Content({body: ''});
+                content.attachments.push(attachment);
+                done(content);
             });
-        });
+        },
+
+        video: function (blob, done) {
+            var url = self._cacheUrl + blob.key;
```
Comment (markdoten) [inline src/javascript/upload/view.js]: This is only used once, I'd say just plug it in instead of having an extra variable.

**Comment by ross-pfahler** [comment]
closing in favor of: https://github.com/Livefyre/streamhub-input/pull/29

**Pull request closed without merging**